airbyte-integrations/connectors/source-smartsheets-2/source_smartsheets_2/source.py

```diff
@@ -0,0 +1,355 @@
+            curr_folder_id, prev_path = folder_stack.pop()
+            logger.debug("requesting folder: '%d'", curr_folder_id)
+            curr_folder = client.Folders.get_folder(curr_folder_id)
+            # Some logic to make paths prettier
+            # 'curr_path' is a tuple of path segments that laters gets converted to a 'pathlib.PurePath'
+            # 'curr_path_str' is the representation that is mainly for logging, can also be used in the metadata
+            if prev_path is None:
+                curr_path = ("/",)
+                curr_path_str = "/"
+            else:
+                curr_path = prev_path + (curr_folder.name,)
+                curr_path_str = curr_path[0] + "/".join(curr_path[1:])
+            logger.info("processing folder: '%s'", curr_path_str)
+
+            # Filter subfolders
+            unmatched_folders, matched_folders = self._filter_folders_by_exclusion(((subfolder, curr_path) for subfolder in curr_folder.folders), exclude_patterns)
+            for subfolder, pattern in matched_folders:
+                logger.debug("subfolder excluded: '%s' -- matched pattern: '%s'", f"{curr_path_str}/{subfolder.name}", pattern)
+            # Reverse iteration to extend the stack in lexical order
+            for subfolder in reversed(unmatched_folders):
+                folder_stack.append((subfolder.id, curr_path))
+
+            # Filter sheets
+            matched_sheets = self._filter_sheets_by_inclusion(((sheet, curr_path) for sheet in curr_folder.sheets), include_patterns)
+            sheet_ids: list[int] = [sheet.id for (sheet, _) in matched_sheets]
+            for sheet, pattern in matched_sheets:
+                logger.debug("sheet included: '%s' -- matched pattern: '%s'", f"{curr_path_str}/{sheet.name}", pattern)
+
+            # Process sheets
+            for sheet_id in sheet_ids:
+                curr_sheet = client.Sheets.get_sheet(sheet_id)
+                logger.info("processing sheet: '%s'", f"{curr_path_str}/{curr_sheet.name}")
+                columns: list[tuple[int, str]] = []
+                # Get whatever columns from the schema are available
+                for col_idx, column in enumerate(curr_sheet.columns):
+                    if column.title in schema_properties:
+                        columns.append((col_idx, column.title))
+                
+                # Process rows with the available columns and the metadata
+                for row in curr_sheet.rows:
+                    metadata: dict[str, Any] = {
+                        field: METADATA_MAPPING[field][1](sheet, row, {"sheet_path": curr_path_str})
+                        for field in config.get("metadata-fields", [])
+                    }
+                    data: dict[str, Any] = {}
+                    for col_idx, col_name in columns:
+                        data[col_name] = row.cells[col_idx].value
+                    # Sometimes we get columns that are actually empty, skip those
```

Comment:
Do you mean rows here?

Reply:
Yeah. I must have thought of a row as a collection of column values, hence `columns`. I will change this.